text/0000-macro-at-most-once-rep.md

```diff
@@ -0,0 +1,166 @@
+- Feature Name: macro-at-most-once-rep
+- Start Date: 2018-01-17
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+
+Summary
+-------
+
+Add a repetition specifiers to macros to repeat a pattern at most once: `$(pat)?`.  `?` behaves like `+` or `*` but represents at most one repetition of `pat`.
+
+Motivation
+----------
+
+There are two specific use cases in mind:
```

Comment:
Stylistic suggestion:
+ Break into two `##` headings for increased readability.